src/main/java/com/squareup/kotlinpoet/ClassName.kt

```diff
@@ -48,6 +48,23 @@ class ClassName private constructor(
 
   override fun asNonNullable() = ClassName(names, false, annotations)
 
+  override fun toKotlinType(): ClassName = when (packageName()) {
+    "java.lang" -> when (simpleName()) {
```

Comment:
There's probably hundreds to the point where we're not going to be able to maintain a list.